stdlib/src/math.rs

```diff
@@ -132,6 +132,9 @@ mod math {
     #[pyfunction]
     fn log(x: PyObjectRef, base: OptionalArg<ArgIntoFloat>, vm: &VirtualMachine) -> PyResult<f64> {
         let base = base.map(|b| *b).unwrap_or(std::f64::consts::E);
+        if base.is_sign_negative() {
+            return Err(vm.new_value_error("math domain error".to_owned()));
+        }
```

Comment:
This is looking not essentially related to Python 3.12 update. If you want to create a separated PR for these changes, that will be quickly merged.